### PR TITLE
[automation] Fixed default value definition for module configuration parameters

### PR DESCRIPTION
**Pull request by cweitkamp**
- Fixed default value definition for module configuration parameters

The field is called "defaultValue" in `ConfigDescriptionParameterDTO` and was not parsed before.

Related to #2148

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
 					"type": "INTEGER",
 					"label": "Offset",
 					"description": "Today +/- offset days (+1 = tomorrow, -1 = yesterday).",
-					"default": 0,
+					"defaultValue": 0,
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json:13]: Imho, the JSON representation of a config description param should be identical to the XML representation where this attribute is called `default`. The only reason why it is called `defaultValue` in the DTO is that `default` is a reserved keyword in Java and thus cannot be used there.
The better fix would therefore imho be to adjust the JSON parsing.

File: bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/GenericConditions.json [hunk 2]
```diff
@@ -66,7 +66,7 @@
 					"type": "TEXT",
 					"description": "the compare operator, allowed are <,>,=,!=,>=,<= matches",
 					"required": true,
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/GenericConditions.json:68]: I wonder why we have required set to true here, if we have a default value defined. According to our usual logic, it should be false - same for other occurrences here in this PR (I know, it was like that before, but I am just seeing it now).

File: bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/ItemConditions.json [hunk 3]
```diff
@@ -18,8 +18,6 @@
 					"type": "TEXT",
 					"label": "Operator",
 					"description": "the compare operator (one of =,<,>,!=,>=,<=)",
-					"required": false,
-					"limitToOptions": true,
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/ItemConditions.json:22]: Why did you remove this attribute?

**Review by kaikreuzer — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json:13]: We can add the `@SerializedName` annotation in the ConfigDescriptionParameterDTO` for this field. e.g.

```java
@SerializedName(value="default", alternate="defaultValue")
```

The first one ("value") will be used as a default for serialization and deserialization, while the second ("defaultValue") will be used to indicate an alternate name of the JSON field that we want to map. But in this way we will introduce a regression as it will be used for exposing configuration description via REST interface too. Which name should be used for exposing the DTO?

Comment (kaikreuzer) [reply bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json:13]: @ghys What's your opinion on the breaking change? Is this something that can be easily adjusted in the Main UI?
It seems to me as half of our config descriptions might not work since we do not use the same attribute name in all places. As it is an optional one, I'd actually hope that the change won't really cause issues in the UI, it might just not recognize the "other half" config descriptions, which currently use `defaultValue` in their JSON.

Comment (ghys) [reply bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json:13]: I count only 4 (relevant) occurrences of `defaultValue` in the UI's code, 2 here:
https://github.com/openhab/openhab-webui/blob/c71db73a6163026f383d352ace8362bc41ccd426/bundles/org.openhab.ui/web/src/components/config/config-sheet.vue#L88
and 2 related to https://github.com/openhab/openhab-webui/pull/1049.
So overall it's feasible but might break a few widgets that make use of the feature in the above PR (which is relatively recent so no big impact anyways).

Comment (kaikreuzer) [reply bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/EphemerisConditions.json:13]: Thanks for checking - should be a rather safe change then.

**Comment by cweitkamp** [comment]
I will close this PR and submit a new one.
